### PR TITLE
Move `Input` loading and dumping APIs from `Testcase` to `Corpus`

### PR DESCRIPTION
**Pull request by domenukk**
This PR contains the comments from #1191, cc @SpaceWhite

### CODE REVIEW

File: libafl/src/corpus/inmemory_ondisk.rs [hunk 1]
```diff
@@ -215,32 +215,30 @@ where
...
                 }
 
-                file = format!("{file_orig}-{ctr}");
+                filename = format!("{filename_orig}-{ctr}");
                 ctr += 1;
             };
 
-            let file_path = self.dir_path.join(filename.clone());
-            let filename_str = file_path.to_str().expect("Invalid Path");
-            testcase.set_filename(filename_str.into())?;
```
Comment (domenukk) [inline libafl/src/corpus/inmemory_ondisk.rs:240]: @SpaceWhite `set_filename` on a full (stringified) path was a bug, right

Comment (SpaceWhite) [reply libafl/src/corpus/inmemory_ondisk.rs:240]: Yes I made a mistake. It should be the name of the testcase instead of full path

Comment (tokatoka) [reply libafl/src/corpus/inmemory_ondisk.rs:240]: the problem is testcase.load_input() & testcase.store_input() assumes they are path but not name

Comment (domenukk) [reply libafl/src/corpus/inmemory_ondisk.rs:240]: Should be addressed now

File: fuzzers/fuzzbench_text/Cargo.toml [hunk 2]
```diff
@@ -20,14 +20,15 @@ cc = { version = "1.0", features = ["parallel"] }
 which = { version = "4.0.2" }
 
 [dependencies]
-libafl = { path = "../../libafl/" }
+libafl = { path = "../../libafl/", features = ["errors_backtrace"] }
```
Comment (tokatoka) [inline fuzzers/fuzzbench_text/Cargo.toml:23]: this was for test?

File: libafl/src/mutators/encoded_mutations.rs [hunk 3]
```diff
@@ -332,13 +332,12 @@ where
...
 
-        let other_size = state
-            .corpus()
-            .get(idx)?
-            .borrow_mut()
-            .load_input()?
-            .codes()
-            .len();
+        let other_size = {
+            let mut other_testcase = state.corpus().get(idx)?.borrow_mut();
```
Comment (andreafioraldi) [inline libafl/src/mutators/encoded_mutations.rs:336]: this pattern seems quite common, @domenukk maybe try to add a method to State to do that like we did for the metadata

**Review by andreafioraldi — DISMISSED** [review]

File: libafl/src/corpus/testcase.rs [hunk 4]
```diff
@@ -322,25 +288,46 @@ impl<I> Testcase<I>
 where
     I: Input + HasLen,
 {
-    /// Get the cached len
+    /// Get the cached `len`. Will `Error::EmptyOptional` if `len` is not yet cached.
     #[inline]
     pub fn cached_len(&mut self) -> Result<usize, Error> {
```
Comment (tokatoka) [inline libafl/src/corpus/testcase.rs:293]: cached_len and load_len is almost the same now.
I think it's better to change cached_len to a simple getter method

Comment (domenukk) [reply libafl/src/corpus/testcase.rs:293]: Good point, done

File: fuzzers/fuzzbench_text/Cargo.toml (hunk 2 above)
Comment (domenukk) [reply fuzzers/fuzzbench_text/Cargo.toml:23]: Removed it again

File: fuzzers/fuzzbench_text/Cargo.toml [hunk 5]
```diff
@@ -28,6 +28,7 @@ clap = { version = "4.0", features = ["default"] }
 nix = "0.26"
 mimalloc = { version = "*", default-features = false }
 content_inspector = "0.2.4"
+#log = "0.4"
```
Comment (SpaceWhite) [inline fuzzers/fuzzbench_text/Cargo.toml:31]: is it better to have optional dependency for this?

https://doc.rust-lang.org/nightly/cargo/reference/features.html#optional-dependencies

File: libafl/src/mutators/encoded_mutations.rs (hunk 3 above)
Comment (domenukk) [reply libafl/src/mutators/encoded_mutations.rs:336]: Cleaned this code up, thanks for the suggestion

File: fuzzers/fuzzbench_text/Cargo.toml (hunk 5 above)
Comment (domenukk) [reply fuzzers/fuzzbench_text/Cargo.toml:31]: It's just there as a reminder for people (mainly me) that want to enable logging in the future

**Pull request merged**